service/src/main/java/ru/karalionak/rest/service/impl/CertificateServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package ru.karalionak.rest.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import ru.karalionak.rest.dao.CertificateDao;
+import ru.karalionak.rest.dao.TagDao;
+import ru.karalionak.rest.entity.Certificate;
+import ru.karalionak.rest.entity.Tag;
+import ru.karalionak.rest.query.CertificateColumns;
+import ru.karalionak.rest.query.Direction;
+import ru.karalionak.rest.query.OrderedViewQuery;
+import ru.karalionak.rest.query.SortingContext;
+import ru.karalionak.rest.service.CertificateService;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class CertificateServiceImpl implements CertificateService {
+    private final CertificateDao certificateDao;
+    private final TagDao tagDao;
+
+    @Autowired
+    public CertificateServiceImpl(CertificateDao certificateDao, TagDao tagDao) {
+        this.certificateDao = certificateDao;
+        this.tagDao = tagDao;
+    }
+
+    @Override
+    @Transactional
+    public void create(Certificate certificate) {
+        long generatedId = certificateDao.create(certificate);
+        Set<Tag> tags = certificate.getTags();
+        createCertificateTagRef(generatedId, tags);
+    }
+
+    @Override
+    @Transactional
+    public void update(long certificateId, Certificate certificate) {
+        certificateDao.update(certificate);
```

Comment:
certificateDao.update(certificate) will fail as you dont provide id to identify what to update.